### DISCUSSION TITLE
Split big next.js, static, production deploy

### DISCUSSION THREAD

**Discussion by sebsonjura** [vercel/next.js#33146] [Ideas]
### Describe the feature you'd like to request

We have a page that has a bout 10-20K static files generated and we have problems with syncing them to our S3 buckets. 

Our page generates a folder called /learn and has a lot of subsequent  pages stored in it 

```
/lernen/first-of-4k-learn-units
``` 

and my question is, if we can split those in next.js in some way, then deploy them to subdomains and get the client to request it from a subdomain 

example 

```
https://f.content.purgae.com/lernen/first-of-4k-learn-units
```

### Describe the solution you'd like

I would like to be able to define a 'splitting pattern' that results in 

* different output folders  by splitting pattern 
* a mechanic in the client to load this from the correct subdomain

### Describe alternatives you've considered

Alternatives considered: 

* SSR (we dont do that)
* removing all static contents out of the page

**Comment by balazsorban44** [comment]
This sounds like a use case for [Incremental Static Regeneration (ISR)](https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration)

Essentially, you wouldn't build the static pages at build-time, but rather run-time. The first visiting the page would get an initial loading screen, but all subsequent users will receive the static page instantly. The pages will keep regenerating depending on the `revalidate` value in `getStaticProps`

**Reply by sebsonjura** [reply]
we do serve only static files. There is no dynamic infrastructure on our side. Just static files served from s3 through cloudfront.
